packages/core-sdk/src/utils/feeUtils.ts

```diff
@@ -276,5 +277,20 @@ export const contractCallWithFees = async ({
     wallet,
     calls,
   });
-  return handleTxOptions({ rpcClient, txOptions, txHash });
+  return handleTransactionResponse(txHash, rpcClient, txOptions) as ContractCallWithFeesResponse<T>;
+};
+
+export const handleTransactionResponse = <T extends Hash | Hash[] = Hash>(
```

Comment:
lets do both `handleTransactionResponse` and `handleTransactionsResponse`. It won't be too much extra code since `handleTransactionResponse` can just call `handleTransactionsResponse`. I think it will make the code more maintainable.